### PR TITLE
Do not register MethodHandleNatives

### PR DESCRIPTION
**Pull request by AntonKozlov**
A simple solution for the lambda problem in the CRaC Core: do not register CALL_SITES at all.

A second part of the PR introduces an interface for Cleaner.register() with priority parameter.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * @rvansa (no known openjdk.org user name / role) ⚠️ Review applies to [dc83f2be](https://git.openjdk.org/crac/pull/76/files/dc83f2be3de189295075fceda75939de0df42cf7)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac.git pull/76/head:pull/76` \
`$ git checkout pull/76`

Update a local copy of the PR: \
`$ git checkout pull/76` \
`$ git pull https://git.openjdk.org/crac.git pull/76/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 76`

View PR using the GUI difftool: \
`$ git pr show -t 76`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/76.diff">https://git.openjdk.org/crac/pull/76.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/crac/pull/76#issuecomment-1562808665)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java [hunk 1]
```diff
@@ -90,8 +90,11 @@ static CallSiteContext make(CallSite cs) {
             // become unreachable, its Context is retained by the Cleanable instance
             // (which is referenced from Cleaner instance which is referenced from
             // CleanerFactory class) until cleanup is performed.
-            new CleanerImpl.PhantomCleanableRef(cs, CleanerFactory.cleaner(),
-                    newContext, JDKResource.Priority.CALL_SITES);
+            // This PhantomCleanableRef is not registered in any Context as
+            // registration cauesed by the core CRaC code leads to deadlock.
```
Comment (DanHeidinga) [inline src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java:94]:
```suggestion
            // registration caused by the core CRaC code leads to deadlock.
```

**Comment by AntonKozlov** [comment]
Tests fail by independent reason (see #77)

File: src/java.base/share/classes/java/lang/ref/Cleaner.java [hunk 2]
```diff
@@ -221,6 +221,15 @@ public Cleanable register(Object obj, Runnable action) {
         return new CleanerImpl.PhantomCleanableRef(obj, this, action, JDKResource.Priority.CLEANERS);
     }
 
+    /**
+     * Register an object and object and also register the underlying Reference with a CRaC priority.
```
Comment (rvansa) [inline src/java.base/share/classes/java/lang/ref/Cleaner.java:225]: an object and action

**Review by rvansa — APPROVED** [review]
LGTM, except the few typos and naming.

**Pull request closed without merging**